modules/game/src/main/BSONHandlers.scala

```diff
@@ -94,7 +96,15 @@ object BSONHandlers {
         castleLastMoveTime = r.get[CastleLastMoveTime](castleLastMoveTime)(CastleLastMoveTime.castleLastMoveTimeBSONHandler),
         unmovedRooks = r.getO[UnmovedRooks](unmovedRooks) | UnmovedRooks.default,
         daysPerTurn = r intO daysPerTurn,
-        clockHistory = ClockHistory(r bytesO moveTimes),
+        binaryMoveTimes = r bytesO moveTimes,
+        clockHistory = for {
+          clk <- gameClock
+          start = clk.limit.seconds
+          ew = (clk.remainingTime(White) * 1000).toLong.millis
+          eb = (clk.remainingTime(Black) * 1000).toLong.millis
+          bw <- r bytesO whiteClockHistory
+          bb <- r bytesO blackClockHistory
+        } yield BinaryFormat.clockHistory.read(start, ew, eb, bw, bb, r intD startedAtTurn, nbTurns),
```

Comment:
Reconsidering this: @isaacl got the decode down to 400ns. During play we will always have to decode anyway. After a move we encode for the DB, but if we make a `case class ... (..., binaryRepr, ...) { lazy val = ...}` we will have to decode again from the binaryRepr instead of reusing the vector. Even this adds some complexity.

All in all I don't think it's worth it.